File: API/APIMeow/Models/Metas.cs


using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

public class Metas
{
    [Key]
    public int IdMeta { get; set; }
    public string Nome { get; set; }
    public decimal GastoLimite { get; set; }
    public int QtsMoedas { get; set; }
    public DateTime DataCriacao { get; set; }
    public DateTime DataTermino { get; set; }
    
    public char Feita{ get; set; }

    [ForeignKey("Usuario")]
    public int IdUsuario { get; set; }
    public Usuario Usuario { get; set; }

    [ForeignKey("Classificacao")]
    public int IdClassificacao { get; set; }
    public Classificacao Classificacao { get; set; }
}

File: front_meow/lib/model/metas.dart
class Metas {
  int idMeta;
  String nome;
  double gastoLimite;
  int qtsMoedas;
  DateTime dataCriacao;
  DateTime dataTermino;
  String feita;
  int idUsuario;
  int idClassificacao;

  Metas({
    required this.idMeta,
    required this.nome,
    required this.gastoLimite,
    required this.qtsMoedas,
    required this.dataCriacao,
    required this.dataTermino,
    required this.feita,
    required this.idUsuario,
    required this.idClassificacao,
  });

  factory Metas.fromJson(Map<String, dynamic> json) {
    return Metas(
      idMeta: json['idMeta'],
      nome: json['nome'],
      gastoLimite: json['gastoLimite'],
      qtsMoedas: json['qtsMoedas'],
      dataCriacao: DateTime.parse(json['dataCriacao']),
      dataTermino: DateTime.parse(json['dataTermino']),
      feita: json['feita'],
      idUsuario: json['idUsuario'],
      idClassificacao: json['idClassificacao'],
    );
  }
}
